rancher/distros-test-framework
# Cert rotate test

#### Proposed Changes ####



Automating the certificate rotate testcase
Added the following: 
1) Helper methods for stop/start services - rke2 or k3s and certificate rotate operation in shared/cluster.go
2) testcase for running the cert rotate test in pkg/testcase/certificates.go
3) Separate Entrypoint/Test Suite for cert rotate: entrypoint/certrotate
4) changes to Makefile and test_runner.sh to call the certrotate tests. 
5) Added docs/test_related.md to convey information regarding setup assumptions for this testcase to be successful



#### Types of Changes ####


Patch Validation testcase

#### Testing ####


Checklist:
1. If your PR changes anything on or related to Jenkins, run it pointing to your branch to make sure it's okay.
N/A.

2. Verify code lint; we should not have errors.
Done

3. Update the documentation if needed.
Added a test_related.md section for any particular test related information. 

4. Update makefile and docker run if adding new tests.
Done

5. Run your tests at least 4 times with all configurations needed and possible.
Done

6. If needed test with different os types.
Not OS related. Tested with sles

#### Linked Issues ####


https://github.com/rancher/qa-tasks/issues/597
https://github.com/rancher/qa-tasks/issues/598

#### Further Comments ####
